java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/Instance.java

```diff
@@ -82,15 +82,19 @@ protected void afterExecute(Runnable t, Throwable e) {
         private final ThreadObserverHelper _observerHelper;
     }
 
+    // The thead pool executor uses an unbounded queue. The tasks would wait in the queue until a
+    // core pool thread is available to run it and the value of the maximumPoolSize therefore
+    // doesn't have any effect.
     private static class QueueExecutor extends java.util.concurrent.ThreadPoolExecutor {
         QueueExecutor(Properties props, String threadName) {
             super(
-                    1,
-                    1,
-                    0,
+                    10,
+                    10,
+                    1000,
```

Comment:
what do these magic numbers refer to?

Reply:
I would add a comment, but see https://docs.oracle.com/javase/9/docs/api/java/util/concurrent/ThreadPoolExecutor.html#ThreadPoolExecutor-int-int-long-java.util.concurrent.TimeUnit-java.util.concurrent.BlockingQueue-

Reply:
Ok, so the `1000` for `keepAliveTime` will have  no effect here since we're setting core and max to 10.